src/pontibus/protocols/relative/hybridtop_units.py

```diff
@@ -0,0 +1,573 @@
+
+from pontibus.protocols.relative.settings import HybridTopProtocolSettings
+from pontibus.protocols.solvation.base import _get_and_charge_solvent_offmol
+from pontibus.utils.settings import InterchangeFFSettings, PackmolSolvationSettings
+from pontibus.utils.system_creation import (
+    _check_and_deduplicate_charged_mols,
+    _get_force_field,
+    interchange_packmol_creation,
+)
+
+
+class HybridTopProtocolUnit(RelativeHybridTopologyProtocolUnit):
+    @staticmethod
+    def _get_interchanges(
+        small_mols,
+        protein_component,
+        solvent_component,
+        forcefield_settings,
+        solvation_settings,
+        charge_settings,
+    ):
+        # Create an smc comp dictionary for stateA
+        stateA_smc_comps = dict(chain(small_mols["stateA"], small_mols["both"]))
+
+        # Get solvent offmol if necessary
+        if solvent_component is not None:
+            solvent_offmol = _get_and_charge_solvent_offmol(
+                solvent_component,
+                solvation_settings,
+                charge_settings,
+            )
+        else:
+            solvent_offmol = None
+
+        # Get the stateA interchange
+        with without_oechem_backend():
+            interA, comp_residsA = interchange_packmol_creation(
+                ffsettings=forcefield_settings,
+                solvation_settings=solvation_settings,
+                smc_components=stateA_smc_comps,
+                protein_component=protein_component,
+                solvent_component=solvent_component,
+                solvent_offmol=solvent_offmol,
+            )
+
+        # We assume a single alchemical component, which this Protocol asserts
+        molA = small_mols["stateA"][0][1]
+        molB = small_mols["stateB"][0][1]
+
+        # Get the molA idx
+        molA_idx = None
+        for idx, mol in enumerate(interA.topology._molecules):
+            if mol.is_isomorphic_with(molA):
+                if np.allclose(mol.conformers[0], molA.conformers[0]):
+                    if molA_idx is not None:
+                        raise ValueError("equality clash with molA")
+                    molA_idx = idx
+                    molA_resnum = mol.atoms[0].metadata["residue_number"]
+
+        if molA_idx is None:
+            raise ValueError("Could not find alchemical molecule A in system creation")
+
+        # Set molB residue number to molA
+        for atom in molB.atoms:
+            atom.metadata["residue_number"] = molA_resnum
+
+        top_mols = [m for m in interA.topology.molecules]
+        top_mols.pop(molA_idx)
+        top_mols.append(molB)
+
+        ff = _get_force_field(forcefield_settings)
+        charged_mols = _check_and_deduplicate_charged_mols(
+            molecules=[m for m in top_mols if m.partial_charges is not None]
+        )
+
+        interB = ff.create_interchange(
+            topology=Topology.from_molecules(top_mols),
+            charge_from_molecules=charged_mols,
+        )
```

Comment:
Trying to extract the final partial charges from molecules in `interA.topology.molecules` is dangerous (a fact which we helpfully haven't documented). It also doesn't save anything - modulo conformer dependence in AM1BCC, assigning partial charges using librarycharges or NAGL is robustly reproducible. So this trick can be dropped altogether. 

```suggestion
        ff = _get_force_field(forcefield_settings)

        interB = ff.create_interchange(
            topology=Topology.from_molecules(top_mols),
            charge_from_molecules=[molB],
        )
```

Reply:
Ahhhh, unless the solvent might not have the same charges as the FF used for the other components. Will that ever be the case? If this protocol is just for OpenFF use then all components will always use the same FF.

Reply:
Solvent is unlikely to be an issue (unless anyone feels like doing some protein-in-toluene experiments), but cofactors are something to consider.

I can just collect these from the input molecules further up through.